### PR TITLE
chore: reimplementation of app stats

### PR DESCRIPTION
**Pull request by gastonfournier**
## About the changes
App stats is mainly used to cap the number of applications reported to Unleash based on the last 7 days information: https://github.com/Unleash/unleash/blob/cc2ccb11342c8f7ba55830d468cad83ed32913a9/src/lib/middleware/response-time-metrics.ts#L24-L28

Instead of getting all stats, just calculate appCount statistics

Use scheduler service instead of setInterval

### CODE REVIEW

File: src/lib/features/instance-stats/instance-stats-service.ts [hunk 1]
```diff
@@ -165,14 +163,9 @@ export class InstanceStatsService {
...
     }
 
-    async refreshStatsSnapshot(): Promise<void> {
+    async refreshAppCountSnapshot(): Promise<void> {
         try {
-            this.snapshot = await this.getStats();
-            const appCountReplacement = {};
-            this.snapshot.clientApps?.forEach((appCount) => {
-                appCountReplacement[appCount.range] = appCount.count;
-            });
```
Comment (gastonfournier) [inline src/lib/features/instance-stats/instance-stats-service.ts:174]: Avoid querying all statistics just to get the appCount. Note that the property `this.snapshot` was never used.

File: src/lib/features/instance-stats/instance-stats-service.ts [hunk 2]
```diff
@@ -279,42 +270,27 @@ export class InstanceStatsService {
...
+        const [t7d, t30d, allTime] = await Promise.all([
+            this.clientInstanceStore.getDistinctApplicationsCount(7),
+            this.clientInstanceStore.getDistinctApplicationsCount(30),
+            this.clientInstanceStore.getDistinctApplicationsCount(),
+        ]);
+        return {
+            '7d': t7d,
+            '30d': t30d,
+            allTime,
+        };
```
Comment (gastonfournier) [inline src/lib/features/instance-stats/instance-stats-service.ts:293]: Use my improved typescript skills

File: src/lib/features/instance-stats/instance-stats-service.ts [hunk 3]
```diff
@@ -251,7 +252,7 @@ export class InstanceStatsService {
             this.strategyStore.count(),
             this.hasSAML(),
             this.hasOIDC(),
-            this.getLabeledAppCounts(),
+            this.appCount ? this.appCount : this.refreshAppCountSnapshot(),
```
Comment (gastonfournier) [inline src/lib/features/instance-stats/instance-stats-service.ts:255]: If already in snapshot use the snapshot

File: src/lib/server-impl.ts [hunk 4]
```diff
@@ -60,7 +60,6 @@ async function createApp(
             await stopServer();
         }
         services.schedulerService.stop();
-        metricsMonitor.stopMonitoring();
```
Comment (gastonfournier) [inline src/lib/server-impl.ts:63]: Scheduler service stop takes care now

File: src/lib/features/instance-stats/instance-stats-service.ts (hunk 2 above)
Comment (ivarconr) [reply src/lib/features/instance-stats/instance-stats-service.ts:303]: I think this is just JavaScript ;)

File: src/lib/features/instance-stats/instance-stats-service.ts [hunk 5]
```diff
@@ -287,34 +288,19 @@ export class InstanceStatsService {
...
-            },
-        ];
+        const [t7d, t30d, allTime] = await Promise.all([
+            this.clientInstanceStore.getDistinctApplicationsCount(7),
+            this.clientInstanceStore.getDistinctApplicationsCount(30),
+            this.clientInstanceStore.getDistinctApplicationsCount(),
+        ]);
+        return {
+            '7d': t7d,
+            '30d': t30d,
```
Comment (ivarconr) [inline src/lib/features/instance-stats/instance-stats-service.ts:301]: Rename t7d to 7d and t30d to 30d?

Comment (gastonfournier) [reply src/lib/features/instance-stats/instance-stats-service.ts:301]: I'd love to, but it's not syntactically correct. Identifiers can't start with numbers.

File: src/lib/routes/admin-api/instance-admin.ts [hunk 6]
```diff
@@ -84,11 +84,11 @@ class InstanceAdminController extends Controller {
...
-            clientApps: [
-                { range: 'allTime', count: 15 },
-                { range: '30d', count: 9 },
-                { range: '7d', count: 5 },
-            ],
+            clientApps: {
+                allTime: 15,
+                '30d': 9,
+                '7d': 5,
+            },
```
Comment (chriswk) [inline src/lib/routes/admin-api/instance-admin.ts:91]: Is the frontend prepared for this change?

Comment (chriswk) [reply src/lib/routes/admin-api/instance-admin.ts:91]: If we use this in the frontend though.

Comment (gastonfournier) [reply src/lib/routes/admin-api/instance-admin.ts]: [d6fa7cf](https://github.com/Unleash/unleash/pull/6155/commits/d6fa7cffeb017972ed59bf8cd052340486ece512) should fix this

**Pull request merged**